src/builder.rs

```diff
@@ -70,26 +70,13 @@ pub enum BuildError {
     CircuitError(#[from] circuit_builder::CircuitBuildError),
 }
 
-impl AsMut<HugrMut> for HugrMut {
-    fn as_mut(&mut self) -> &mut HugrMut {
-        self
-    }
-}
-impl AsRef<HugrMut> for HugrMut {
-    fn as_ref(&self) -> &HugrMut {
-        self
-    }
-}
-
-/// Trait allowing treating type as (im)mutable reference to [`HugrMut`]
-pub trait HugrMutRef: AsMut<HugrMut> + AsRef<HugrMut> {}
-impl HugrMutRef for HugrMut {}
-impl HugrMutRef for &mut HugrMut {}
+/// Trait allowing treating type as (im)mutable reference to [`Hugr`]
+pub trait HugrMutRef: AsMut<Hugr> + AsRef<Hugr> {}
+impl<H: HugrMut> HugrMutRef for H {}
+impl HugrMutRef for &mut Hugr {}
```

Comment:
This can also be moved with the rest of the logic in `hugr.rs`.

Reply:
HugrMutRef deleted entirely!